src/catalog/catalog_manager.py

```diff
@@ -200,3 +200,50 @@ def get_dataset_metadata(self, database_name: str, dataset_name: str,
 
         return self._dataset_service.dataset_object_by_name(
             database_name, dataset_name)
+
+    def delete_column_metadata(self, table_name: str,
+                           column_names: List[str]) -> DataFrameMetadata:
+        """
+        This method deletes the columns associated with the given
+        metadata
+
+        Arguments:
+           table_name-[str] table for which we will delete the columns
+           column_names - [List of columns that needs to deleted]
+
+        Retuns:
+           The updated DataFrameMetadata object for which the respective columns are deleted
```

Comment:
I think we should remove this return part. Let us stick to just deleting the required columns.